orca-clouddriver/src/main/groovy/com/netflix/spinnaker/orca/clouddriver/tasks/snapshot/DeleteSnapshotTask.java

```diff
@@ -0,0 +1,103 @@
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.netflix.spinnaker.orca.clouddriver.tasks.snapshot;
+
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import java.util.concurrent.TimeUnit;
+import java.util.stream.Collectors;
+import javax.annotation.Nonnull;
+import javax.validation.ConstraintViolation;
+import javax.validation.Validation;
+import com.netflix.spinnaker.orca.ExecutionStatus;
+import com.netflix.spinnaker.orca.RetryableTask;
+import com.netflix.spinnaker.orca.TaskResult;
+import com.netflix.spinnaker.orca.clouddriver.KatoService;
+import com.netflix.spinnaker.orca.clouddriver.model.TaskId;
+import com.netflix.spinnaker.orca.clouddriver.pipeline.snapshot.DeleteSnapshotStage;
+import com.netflix.spinnaker.orca.clouddriver.tasks.AbstractCloudProviderAwareTask;
+import com.netflix.spinnaker.orca.pipeline.model.Stage;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+import static java.util.stream.Collectors.toList;
+
+@Component
+public class DeleteSnapshotTask extends AbstractCloudProviderAwareTask implements RetryableTask {
+  private final KatoService katoService;
+
+  @Autowired
+  public DeleteSnapshotTask(KatoService katoService) {
+    this.katoService = katoService;
+  }
+
+  @Override
+  public TaskResult execute(@Nonnull Stage stage) {
+    DeleteSnapshotStage.DeleteSnapshotRequest deleteSnapshotRequest = stage.mapTo(DeleteSnapshotStage.DeleteSnapshotRequest.class);
+    validateInputs(deleteSnapshotRequest);
+
+    List<Map<String, Map>> operations = deleteSnapshotRequest
+      .getSnapshotIds()
+      .stream()
+      .map(snapshotId -> {
+        Map<String, Object> operation = new HashMap<>();
+        operation.put("credentials", deleteSnapshotRequest.getCredentials());
+        operation.put("region", deleteSnapshotRequest.getRegion());
+        operation.put("snapshotId", snapshotId);
+
+        Map<String, Map> tmp = new HashMap<>();
+        tmp.put("deleteSnapshot", operation);
+        return tmp;
+      }).collect(toList());
+
+    TaskId taskId = katoService
+      .requestOperations(deleteSnapshotRequest.getCloudProvider(), operations).toBlocking().first();
+
+    Map<String, Object> outputs = new HashMap<>();
+    outputs.put("notification.type", "deleteSnapshot");
+    outputs.put("kato.last.task.id", taskId);
+    outputs.put("delete.region", deleteSnapshotRequest.getRegion());
+    outputs.put("delete.account.name", deleteSnapshotRequest.getCredentials());
+
+    return new TaskResult(ExecutionStatus.SUCCEEDED, outputs);
+
+  }
+
+  private void validateInputs(DeleteSnapshotStage.DeleteSnapshotRequest deleteSnapshotRequest) {
+    Set<ConstraintViolation<DeleteSnapshotStage.DeleteSnapshotRequest>> violations = Validation.buildDefaultValidatorFactory()
```

Comment:
Not sure this is worth it ... we should just let `clouddriver` do the validation and avoid this sort of thing in `orca`.

Reply:
removed the validation from orca